torch/package/importer.py

```diff
@@ -239,6 +253,12 @@ def _gcd_import(self, name, package=None, level=0):
         the loader did not.
 
         """
+        if name == self._mangler.parent_name():
```

Comment:
I don't get this part -- something is wrong if the import system is trying to import a package with the mangled prefix still attached. This indicates that something didn't demangle a name before using it.

Reply:
hm yeah—my initial thought was that `_gcd_import` was the right place to put the mangling logic (since it is the greatest common denominator of all the import pathways) but I think it's actually more confusing compared to demangling at the highest level.